File: Algorithms/Sieve of Eratosthenes/sieve_of_eratosthenes.py
def eratosthenes_sieve(limit):
    sieve = [True] * (limit+1)                      # Initialize the primality list
    sieve[0] = sieve[1] = False
    result = []                                     # Result list of prime numbers in given range
    for i in range(2, limit+1):
        if sieve[i]:
            result.append(i)                        # append primary number to result list
            for k in range(2, int(limit/i)+1):      # mark folowing non-primary numbers 
                sieve[i*k] = False
    return result                                   # yield the resulting list of primary numbers

# print eratosthenes_sieve(101)

File: Algorithms/Sieve of Eratosthenes/sieve_of_eratosthenes.cpp
#include <bits/stdc++.h>
using namespace std;

int prime[100001]={0};

int sieve()
{
    int i,j;
    prime[0]=1;
    prime[1]=1;
    for(i=2;i*i<=100000;i++)
    {
        if(prime[i]==0)
        {
            for(j=i*i;j<=100000;j+=i)
                prime[j]=1;
        }
    }
}

int main()
{
    int i,t,l,r;
    cout<<"Enter the no of test ranges:";
    cin>>t;
    sieve();
    while(t--)
    {
        cout<<"Enter the range:";
        cin>>l;
        cin>>r;
        for(i=l;i<=r;i++)
        {
            if(prime[i]==0)
                cout<<i<<" ";
        }
        cout<<endl;
    }
    return 0;
}
